sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -154,22 +171,52 @@ def __str__(self):
         return event_str
 
     @classmethod
-    def _from_message(cls, message):
-        # type: (Message) -> EventData
+    def _from_message(cls, message, raw_amqp_message=None):
+        # type: (Message, Optional[AmqpAnnotatedMessage]) -> EventData
         """Internal use only.
 
-        Creates an EventData object from a raw uamqp message.
+        Creates an EventData object from a raw uamqp message and, if provided, AmqpAnnotatedMessage.
 
         :param ~uamqp.Message message: A received uamqp message.
+        :param ~azure.eventhub.amqp.AmqpAnnotatedMessage message: An amqp annotated message.
         :rtype: ~azure.eventhub.EventData
         """
         event_data = cls(body="")
-        event_data.message = message
+        if raw_amqp_message and message:
+            event_data._raw_amqp_message = raw_amqp_message
+            event_data.message = message
+        elif message:
+            event_data._raw_amqp_message = AmqpAnnotatedMessage(message=message)
+            event_data.message = message
```

Comment:
nit: I think this could be improved as the following:
```suggestion
        event_data.message = message
        event_data._raw_amqp_message = raw_amqp_message if raw_amqp_message else AmqpAnnotatedMessage(message=message)
```